src/test/java/ua/bookstore/online/controller/BookControllerIntegrationTest.java

```diff
@@ -0,0 +1,226 @@
+import static ua.bookstore.online.utils.ConstantAndMethod.getNewOrwell;
+import static ua.bookstore.online.utils.ConstantAndMethod.getOrwell;
+import static ua.bookstore.online.utils.ConstantAndMethod.getRequestDto;
+import static ua.bookstore.online.utils.ConstantAndMethod.tearDown;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.List;
+import javax.sql.DataSource;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.http.MediaType;
+import org.springframework.http.ProblemDetail;
+import org.springframework.jdbc.datasource.init.ScriptUtils;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+import org.testcontainers.shaded.org.apache.commons.lang3.builder.EqualsBuilder;
+import ua.bookstore.online.dto.book.BookDto;
+import ua.bookstore.online.dto.book.CreateBookRequestDto;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class BookControllerIntegrationTest {
+    private static final String URI = "/books";
+    private static MockMvc mockMvc;
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @BeforeAll
+    static void beforeAll(@Autowired WebApplicationContext applicationContext) {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+    }
+
+    @BeforeEach
+    void beforeEach(@Autowired DataSource dataSource) throws SQLException {
+        tearDown(dataSource);
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource(ADD_CATEGORIES_SQL));
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource(ADD_THREE_BOOKS_SQL));
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource(ADD_CATEGORIES_FOR_BOOKS_SQL));
+        }
+    }
+
+    @AfterEach
+    void afterEach(@Autowired DataSource dataSource) {
+        tearDown(dataSource);
+    }
+
+    @Test
+    @DisplayName("Create new book")
+    @WithMockUser(username = "admin", roles = {"MANAGER"})
+    void createBook_CreateNewBook_ReturnsExpectedBook() throws Exception {
```

Comment:
The name of the test method does not correspond to what is tested in the method (I see that this test checks that conflict response is received if ISBN is not unique)

Reply:
I do it in the next PR.